backend/app/services/quiz.py

```diff
@@ -10,3 +10,8 @@ async def get_quizzes(session: AsyncSession) -> Sequence[Quiz]:
     statement = select(Quiz).order_by(Quiz.created_at)
     result = await session.execute(statement)
     return result.scalars().all()
+
+async def get_quiz_by_id(session: AsyncSession, quiz_id: int) -> Quiz:
+    statement = select(Quiz).filter(Quiz.id == quiz_id)
+    result = await session.execute(statement)
+    return result.scalars().first()
```

Comment:
I'm not 100% sure, but it can return `None` as well. I'd add checking whether the quiz was returned and raise the HTTPException accordingly.